File: tests/src/com.google.protobuf/protobuf-javalite/3.25.5/src/test/java/com_google_protobuf/protobuf_javalite/ExtensionSchemasTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_google_protobuf.protobuf_javalite;

import com_google_protobuf.protobuf_javalite.fixture.ExtensionSchemasCoverageProto.Proto2Message;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class ExtensionSchemasTest {
    @Test
    void proto2LiteMessagesUseExtensionSchemaDuringSerialization() throws Exception {
        Proto2Message message = Proto2Message.newBuilder()
                .setLabel("extension schema coverage")
                .build();

        byte[] serialized = message.toByteArray();
        Proto2Message parsed = Proto2Message.parseFrom(serialized);

        assertThat(parsed.getLabel()).isEqualTo("extension schema coverage");
    }
}

File: tests/src/org.apache.pekko/pekko-protobuf-v3_2.13/1.1.5/src/test/scala/org_apache_pekko/pekko_protobuf_v3_2_13/ExtensionSchemasTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_protobuf_v3_2_13

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ExtensionSchemasTest {
  @Test
  def buildsExtensionSchemasForProto2LiteMessages(): Unit = {
    val message: ExtensionSchemasProbe.Proto2LiteMessage = ExtensionSchemasProbe.parseProto2LiteMessage()

    assertThat(message).isNotNull
  }
}
